tools/github/release/abstract.py

```diff
@@ -0,0 +1,313 @@
+import pathlib
+from abc import abstractmethod
+from functools import cached_property
+from typing import (
+    Any, AsyncGenerator, Awaitable, Coroutine, Dict, Iterable, Iterator, List, Optional, Pattern,
+    Set, Type, Union)
+
+import verboselogs  # type:ignore
+
+import packaging.version
+
+import aiohttp
+
+import gidgethub.abc
+
+from tools.base import abstract
+from tools.base.functional import async_property
+
+
+class AGithubReleaseAssets(metaclass=abstract.Abstraction):
+    """Base class for Github release assets pusher/fetcher"""
+
+    @abstractmethod
+    def __init__(self, release: "AGithubRelease", path: pathlib.Path) -> None:
+        raise NotImplementedError
+
+    @abstractmethod
+    async def __aiter__(self) -> AsyncGenerator[Dict[str, Union[str, pathlib.Path]], Awaitable]:
+        if False:
```

Comment:
?

Reply:
this method is a generator, generators have to yield - from my searching this is a ~standard way to mark a generator interface